src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double SIZE_MULTIPLIER = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        increaseSize();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkAddIndex(index);
+        increaseSize();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        array[index] = value;
     }
 
     @Override
-    public T remove(int index) {
-        return null;
+    public T remove(int index) throws ArrayListIndexOutOfBoundsException {
+        checkIndex(index);
+        T deletedValue = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        size--;
+        return deletedValue;
```

Comment:
In the `remove(int index)` method, the `size` is decremented before returning the deleted value, which is correct. However, ensure that the `size` is managed properly in all methods to avoid potential `ArrayIndexOutOfBoundsException`.